File: HW/HW11/hw11.py
from typing import List

class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        if (image[sr][sc] == color): 
            return image
        
        self.fill(image, sr, sc, image[sr][sc], color)

        return image
    
    def fill(self, image: List[List[int]], sr: int, sc: int, currColor:int, newColor: int):
        row = len(image)
        col = len(image[0])

        if (sr < 0 or sc < 0 or sr >= row or sc >= col or image[sr][sc] != currColor): 
            return
        
        image[sr][sc] = newColor

        self.fill(image, sr + 1, sc, currColor, newColor)
        self.fill(image, sr - 1, sc, currColor, newColor)
        self.fill(image, sr, sc + 1, currColor, newColor)
        self.fill(image, sr, sc - 1, currColor, newColor)

File: HW/HW11/hw11.java
package HW.HW11;

public class hw11 {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if (image[sr][sc] == color) { 
            return image;
        }
        fill(image, sr, sc, image[sr][sc], color);
        return image;
    }

    public void fill(int[][] image, int sr, int sc, int currentColor, int newColor){ 
        int rowLength = image.length;
        int colLength = image[0].length;
        if (sr < 0 || sc < 0 || sr >= rowLength || sc >= colLength|| image[sr][sc] != currentColor) {
            return;
        }

        image[sr][sc] = newColor;
        fill(image, sr + 1, sc, currentColor, newColor);
        fill(image, sr - 1, sc, currentColor, newColor);
        fill(image, sr, sc + 1, currentColor, newColor);
        fill(image, sr, sc - 1, currentColor, newColor);
    }

    public static void main(String[] args) {
        
    }
}
